### PR TITLE
Add a link for producing issues

### PR DESCRIPTION
**Pull request by oli-obk**
I am very sorry if I am violating every rule imaginable... I don't even know what language this is written in. I just looked at whatever was going on and copied stuff.

### CODE REVIEW

**Comment by shepmaster** [comment]
My immediate thoughts:

1. *Oh my god* this is going to cause a lot of really terrible issues to be created.
1. This is going to have poor optics: "Rust is *so bad* that they had to optimize reporting compiler bugs from the playground that's linked from every documentation site and the front page of the Rust website"
1. People are going to use this to report bugs with the playground into the Rust repository.
1. People are going to use this to report bugs with specific *crates* into the Rust repository.

Is this intended to do anything beyond optimizing the process of copy-and-pasting code out of one textbox and into another?

**Comment by oli-obk** [comment]
> Is this intended to do anything beyond optimizing the process of copy-and-pasting code out of one textbox and into another?

Is that not a valid use case? :stuck_out_tongue_winking_eye: 

> This is going to have poor optics: "Rust is so bad that they had to optimize reporting compiler bugs from the playground that's linked from every documentation site and the front page of the Rust website"

I think it's mainly for diagnostics.

> People are going to use this to report bugs with specific crates into the Rust repository.

That's fine, see the above

> Oh my god this is going to cause a lot of really terrible issues to be created.

At least they'll have an MCVE, which is better than a lot of other issues

> People are going to use this to report bugs with the playground into the Rust repository.

The playpen has bugs? No, for real, I don't think this is going to be an issue, we can be wordier if we want to to make really sure of it.


I have no problem with you closing this issue, it was derived from a drive-by comment by @nikomatsakis . The alternative I see is auto-expanding code snippets to playpen links and diagnostics or playpen links to code snippets and diagnostics via a bot

**Comment by oli-obk** [comment]
That said, I do not have a strong opinion about this feature, feel free to just close!

File: ui/frontend/Output.tsx [hunk 1]
```diff
@@ -140,18 +140,22 @@ interface FormatProps {
...
-const Gist: React.SFC<GistProps> = ({ focus, requestsInProgress, id, url, channel, mode }) => {
+const Gist: React.SFC<GistProps> = ({ focus, requestsInProgress, id, url, channel, mode, code, stderr }) => {
   if (focus === 'gist') {
     const loader = (requestsInProgress > 0) ? <MyLoader /> : null;
     const permalink = id ?
       <p><a href={`/?gist=${id}&version=${channel}&mode=${mode}`}>Permalink to the playground</a></p> : null;
     const directLink = url ? (<p><a href={url}>Direct link to the gist</a></p>) : null;
+    const body = id ? `[Playground Link](https://play.rust-lang.org/?gist=${id}&version=${channel}&mode=${mode})%0A%0A%60%60%60rust${code}%60%60%60%0A%0AOutput:%0A%60%60%60${output}%60%60%60` : null;
+    const issueLink = id ?
+      <p><a href={`https://github.com/rust/rust/issues/new?body=${body}`}>Open an issue on the rustc repository</a></p> : null;
```
Comment (jonas-schievink) [inline ui/frontend/Output.tsx]: You got the organization wrong (this link 404s) - it's "github.com/**rust-lang**/rust"

**Comment by shepmaster** [comment]
Thinking about this some more, what do you think about me extending this a little bit?

We could also allow [posting the code to urlo](https://meta.discourse.org/t/compose-a-new-pre-filled-topic-via-url/28074). That would be the primary choice, with the issue tracker as a second choice. This way, we might be able to reduce some of the false positives.

A *future* enhancement would be to grab the crates in play, see if they have GitHub repository URLs, and offer to report to them...

**Comment by oli-obk** [comment]
Yes, that makes absolute sense! I'll add that.

**Comment by shepmaster** [comment]
> I'll add that.

You sure? It's going to be a sizeable chunk of extra work.

**Comment by oli-obk** [comment]
> You sure? It's going to be a sizeable chunk of extra work.

not the crates part ;) Just the reporting to the forum

**Comment by shepmaster** [comment]
I'll take it from here; there's some tidying up I'd like to do. Is it cool if I force-push to your branch?

**Comment by oli-obk** [comment]
> Is it cool if I force-push to your branch?

I consider the "Allow edits from maintainers" checkbox being set as permission granted ;) So yes.

**Pull request merged**